Marlin/src/module/stepper.h

```diff
@@ -280,30 +234,20 @@ class Stepper {
       static void babystep(const AxisEnum axis, const bool direction); // perform a short step with a single stepper motor, outside of any convention
     #endif
 
-    static inline void kill_current_block() {
-      step_events_completed = current_block->step_event_count;
-    }
-
-    //
-    // Handle a triggered endstop
-    //
-    static void endstop_triggered(const AxisEnum axis);
-
-    //
-    // Triggered position of an axis in mm (not core-savvy)
-    //
-    FORCE_INLINE static float triggered_position_mm(const AxisEnum axis) {
-      return endstops_trigsteps[axis] * planner.steps_to_mm[axis];
-    }
-
     #if HAS_MOTOR_CURRENT_PWM
       static void refresh_motor_power();
     #endif
 
   private:
 
-    FORCE_INLINE static hal_timer_t calc_timer_interval(hal_timer_t step_rate) {
-      hal_timer_t timer;
+    // Set the current position in steps
+    static void _set_position(const int32_t &a, const int32_t &b, const int32_t &c, const int32_t &e);
+
+    // Set direction bits for all steppers
+    static void set_directions();
+
+    FORCE_INLINE static uint32_t calc_timer_interval(uint32_t step_rate) {
+      uint32_t timer;
```

Comment:
Perhaps we should keep using `hal_timer_t` for all the values related to interrupt counters.

We should of course continue to use `uint32_t` for all values related to step counting in blocks, acceleration, and so on. If it happens, for any reason, to be better to use signed `int32_t` for those, we can afford the loss of 1 bit. 2^31 is more than enough range for our purposes.

Reply:
The idea behind using uint32_t was that we emulate a 32bit timer in all platforms (and isr_scheduler() does exactly that. 
But let´s place figures:
On AVR, timer is 16 bits. Running at 16mhz, the base frequency of the timer is 2Mhz. So, the maximum period possible is 1/30 of a second. Assuming an axis has 96pulses per mm, that means that the minimum possible speed is 3mm/second.
If we want to print slower, we NEED a 32bit value.
That is why it makes sense to keep a 32bit timer value there 👍